components/ntp_sponsored_images/browser/ntp_sponsored_image_source.cc

```diff
@@ -44,11 +44,16 @@ std::string NTPSponsoredImageSource::GetSource() {
 }
 
 void NTPSponsoredImageSource::StartDataRequest(
-    const std::string& path,
+    const GURL& url,
     const content::WebContents::Getter& wc_getter,
-    const GotDataCallback& callback) {
+    GotDataCallback callback) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
 
+  std::string path = url.path();
```

Comment:
what are we doing here? This needs some explanation in a comment

Reply:
Added a comment in c0b7bce531033f2ed82772dcad093cc918867ea6. 
Class methods like `IsValidPath` and `IsLogPath`, etc. expect just the path, but GURL::path() returns the path with the leading slash. This gets rid of the slash.